### DISCUSSION TITLE
async_session.get doesn't get joinedload values

### DISCUSSION THREAD

**Discussion by Actticus** [sqlalchemy/sqlalchemy#6815] [Usage Questions]
In my code, I have one session to one case. And before getting an object I need to validate a user (sign in case). 

```
    async def validate(self, session):
        self.phone = (
            await session.execute(
                sqlalchemy.select(self.phone_model)
                .where(
                    sqlalchemy.and_(
                        self.phone_model.number == self.data["phone"],
                        self.phone_model.is_confirmed == True
                    )
                )
                .options(
                    sqlalchemy.orm.joinedload(self.phone_model.patient),
                )
            )
        ).first()[0]

        await self.validate_object_exists(self.phone, field="username")
        await self.validate_phone_has_user(self.phone, field="username")
```

After that, I'm trying to get a Patient

```
    async def authenticate(self, session):
        if self.phone.patient.is_right_password(self.data["password"]):
            patient = await session.get(
                models.Patient,
                self.phone.patient.id,
                options=[
                    sqlalchemy.orm.joinedload(models.Patient.phone),
                    sqlalchemy.orm.joinedload(models.Patient.sex),
                    sqlalchemy.orm.joinedload(models.Patient.diagnoses),
                ]
            )
            return patient
```

I'm checking what I have inside a Patient in debugging mode, and I have joined only a phone. But if I create a new session or skip a validate I get joined all models (phone, sex, diagnoses)

**Comment by CaselIT** [comment]
Hi,

That is the expected behaviour documented here: https://docs.sqlalchemy.org/en/14/orm/session_api.html?highlight=get#sqlalchemy.orm.Session.get

> Session.get() is special in that it provides direct access to the identity map of the Session. If the given primary key identifier is present in the local identity map, the object is returned directly from this collection and no SQL is emitted, unless the object has been marked fully expired

If the session has the requested object in the identity map it does not emit sql. You can override this by setting [`populate_existing=True`](https://docs.sqlalchemy.org/en/14/orm/session_api.html?highlight=get#sqlalchemy.orm.Session.get.params.populate_existing) of by using a `select` + `session.execute()`

**Reply by Actticus** [reply]
Maybe I misunderstand, but logically, if there are not enough loaded models requested, should not emit SQL to load dependencies?

**Reply by CaselIT** [reply]
There you are loading a single `models.Patient` so the check on the identity map is only on it's id, not on it's connected relationships.
